runtime panic

I was testing stn for 40% voting power and hit the following panic.

```
panic: runtime error: index out of range [-1]

goroutine 139 [running]:
github.com/harmony-one/harmony/api/service/syncing.(*SyncConfig).cleanUpPeers(0xc00b294960, 0xffffffffffffffff)
        /home/lc/go/src/github.com/harmony-one/harmony/api/service/syncing/syncing.go:432 +0x1e2
github.com/harmony-one/harmony/api/service/syncing.(*SyncConfig).GetBlockHashesConsensusAndCleanUp(0xc00b294960)
        /home/lc/go/src/github.com/harmony-one/harmony/api/service/syncing/syncing.go:462 +0x170
github.com/harmony-one/harmony/api/service/syncing.(*StateSync).getConsensusHashes(0xc000115200, 0xc00b1d1b40, 0x20, 0x20, 0x431c908b0000002f)
        /home/lc/go/src/github.com/harmony-one/harmony/api/service/syncing/syncing.go:495 +0xd7
github.com/harmony-one/harmony/api/service/syncing.(*StateSync).ProcessStateSync(0xc000115200, 0xc00b1d1b40, 0x20, 0x20, 0x431c908b0000002f, 0xc000a22000, 0xc00097c730, 0xc0015e6180, 0x4)
        /home/lc/go/src/github.com/harmony-one/harmony/api/service/syncing/syncing.go:863 +0x55
github.com/harmony-one/harmony/api/service/syncing.(*StateSync).SyncLoop(0xc000115200, 0xc000a22000, 0xc00097c730, 0x0, 0xc0000d2900)
        /home/lc/go/src/github.com/harmony-one/harmony/api/service/syncing/syncing.go:1016 +0x345
github.com/harmony-one/harmony/node.(*Node).doSync(0xc0000de840, 0xc000a22000, 0xc00097c730, 0x1)
        /home/lc/go/src/github.com/harmony-one/harmony/node/node_syncing.go:273 +0xdc
github.com/harmony-one/harmony/node.(*Node).DoSyncing(0xc0000de840, 0xc000a22000, 0xc00097c730, 0x2601)
        /home/lc/go/src/github.com/harmony-one/harmony/node/node_syncing.go:239 +0x16c
created by github.com/harmony-one/harmony/node.(*Node).SupportSyncing
        /home/lc/go/src/github.com/harmony-one/harmony/node/node_syncing.go:310 +0xca
```

--- a/api/service/syncing/syncing.go
+++ b/api/service/syncing/syncing.go
@@ -447,7 +447,7 @@ func (sc *SyncConfig) cleanUpPeers(maxFirstID int) {
 // Note that choosing the most common peer config does not guarantee that the blocks to be downloaded are the correct ones.
 // The subsequent node syncing steps of verifying the block header chain will give such confirmation later.
 // If later block header verification fails with the sync peer config chosen here, the entire sync loop gets retried with a new peer set.
-func (sc *SyncConfig) GetBlockHashesConsensusAndCleanUp() {
+func (sc *SyncConfig) GetBlockHashesConsensusAndCleanUp() error {
 	sc.mtx.Lock()
 	defer sc.mtx.Unlock()
 	// Sort all peers by the blockHashes.
@@ -459,11 +459,16 @@ func (sc *SyncConfig) GetBlockHashesConsensusAndCleanUp() {
 		Int("maxFirstID", maxFirstID).
 		Int("maxCount", maxCount).
 		Msg("[SYNC] block consensus hashes")
+
+	if maxFirstID == -1 {
+		return errors.New("invalid peer index -1 for block hashes query")
+	}
 	sc.cleanUpPeers(maxFirstID)
+	return nil
 }
 
 // getConsensusHashes gets all hashes needed to download.
-func (ss *StateSync) getConsensusHashes(startHash []byte, size uint32) {
+func (ss *StateSync) getConsensusHashes(startHash []byte, size uint32) error {
 	var wg sync.WaitGroup
 	ss.syncConfig.ForEachPeer(func(peerConfig *SyncPeerConfig) (brk bool) {
 		wg.Add(1)
@@ -492,8 +497,11 @@ func (ss *StateSync) getConsensusHashes(startHash []byte, size uint32) {
 		return
 	})
 	wg.Wait()
-	ss.syncConfig.GetBlockHashesConsensusAndCleanUp()
+	if err := ss.syncConfig.GetBlockHashesConsensusAndCleanUp(); err != nil {
+		return err
+	}
 	utils.Logger().Info().Msg("[SYNC] Finished getting consensus block hashes")
+	return nil
 }
 
 func (ss *StateSync) generateStateSyncTaskQueue(bc *core.BlockChain) {
@@ -860,7 +868,9 @@ func (ss *StateSync) generateNewState(bc *core.BlockChain, worker *worker.Worker
 // ProcessStateSync processes state sync from the blocks received but not yet processed so far
 func (ss *StateSync) ProcessStateSync(startHash []byte, size uint32, bc *core.BlockChain, worker *worker.Worker) error {
 	// Gets consensus hashes.
-	ss.getConsensusHashes(startHash, size)
+	if err := ss.getConsensusHashes(startHash, size); err != nil {
+		return errors.Wrap(err, "getConsensusHashes")
+	}
 	ss.generateStateSyncTaskQueue(bc)
 	// Download blocks.
 	if ss.stateSyncTaskQueue.Len() > 0 {

